src/entity/components/Velocity.ts

```diff
@@ -1,23 +1,26 @@
 import type { ECSComponent } from '@/ecs/ECSComponent';
 import type { ECSEntity } from '@/ecs/ECSEntity';
+import type { Point } from '@/utils/types';
 
-const VelocityBrand = 'velocity';
+export const VelocityBrand = 'velocity';
 type VelocityBrand = typeof VelocityBrand;
 export type Velocity = ECSComponent<
   VelocityBrand,
   {
-    x: number;
-    y: number;
+    speed: number;
+    target: Point;
   }
```

Comment:
I would claim that a velocity is a speed vector. As such, it's just a dx / dy. How does speed relate to target? Is target a unit vector or a "I want to be on that tile"?

Like, we have various forces acting on an entity. Like the player or AI movement direction, env, like mud.